app/controllers/profiles_controller.rb

```diff
@@ -0,0 +1,15 @@
+class ProfilesController < ApplicationController
+  before_action :require_login, :allow_anyone
+
+  def show
+    checkouts = current_user.checkouts
+    checked_out_book_ids = checkouts.pluck(:book_id)
```

Comment:
You don't need to use this method (which I think I had suggested in the previous PR iteration). I had suggested it when you were displaying *all* of the checkout out books and then trying to add a "return" link if the book was checked out by the current user. Since this controller action and view only shows books the user has checked out, you can do things more conventionally or directly.

You've already got all of the users checkout records with `current_user.checkouts`. You can access the book directly in the view from there when iterating like so:

```
<% @checkouts.each do |checkout| %>
  <li> <%= checkout.book.title %></li>
<% end %>
```

If you can update the code to do this, I think it will simplify this controller action some.